diff -Nur openivr-1.13.5/main.cxx openivr/main.cxx
--- openivr-1.13.5/main.cxx	2003-02-26 02:30:40.000000000 +0100
+++ openivr/main.cxx	2004-06-30 14:27:28.000000000 +0200
@@ -24,6 +24,24 @@
  * Contributor(s): ______________________________________.
  *
  * $Log$
+ * Revision 1.10  2004/06/30 12:27:28  rjongbloed
+ * Rewrite of plug in system to use single global variable for all factories to avoid all sorts
+ *   of issues with startup orders and Windows DLL multiple instances.
+ *
+ * Revision 1.9  2004/06/22 11:01:13  csoutheren
+ * Updated for changed to TTS routines
+ *
+ * Revision 1.8  2004/06/03 12:27:12  rjongbloed
+ * Updated to new factory based capability system.
+ *
+ * Revision 1.7  2004/05/03 13:25:46  rjongbloed
+ * Converted everything to be codec plug in freindly
+ * Removed GSM and G.729 as now plug ins are "the way"!
+ *
+ * Revision 1.6  2004/04/06 11:27:50  rjongbloed
+ * Changes to support native C++ Run Time Type Information
+ * Changes for codec plug ins
+ *
  * Revision 1.5  2003/02/26 01:30:40  robertj
  * Fixed bandwidth problem and ability to listen on specific port, thansk Ivanov
  *
@@ -359,10 +377,10 @@
     localCapabilities.SetCapability(0, 0, new G7231_File_Capability);
 
   // add all other capabilities
-  localCapabilities.SetCapability(0, 0, new H323_GSM0610Capability);
-  localCapabilities.SetCapability(0, 0, new MicrosoftGSMAudioCapability);
-  localCapabilities.SetCapability(0, 0, new H323_G711Capability(H323_G711Capability::ALaw, H323_G711Capability::At64k));
-  localCapabilities.SetCapability(0, 0, new H323_G711Capability(H323_G711Capability::muLaw, H323_G711Capability::At64k));
+  localCapabilities.AddAllCapabilities(0, 0, OpalGSM0610);
+  localCapabilities.AddAllCapabilities(0, 0, "MS-GSM");
+  localCapabilities.AddAllCapabilities(0, 0, OpalG711uLaw);
+  localCapabilities.AddAllCapabilities(0, 0, OpalG711ALaw);
 
   // add user input capabilities
   H323_UserInputCapability::AddAllCapabilities(localCapabilities, 0, 0);
@@ -428,7 +446,7 @@
     if (rtpSession == NULL) {
       PTRACE(4, "H323VOICE\tNo RTP session found");
     } else {
-      if (!rtpSession->IsDescendant(RTP_UDP::Class())) {
+      if (!PIsDescendant(rtpSession, RTP_UDP)) {
         PTRACE(4, "H323VOICE\tRTP session is not descendant of RTP_UDP");
       } else
         srcAddress = ((RTP_UDP *)rtpSession)->GetRemoteAddress().AsString();
@@ -460,7 +478,16 @@
 IVRVXMLSession::IVRVXMLSession(OpenIVR & _app, MyH323Connection * _conn)
   : OpalVXMLSession(_conn), app(_app)
 {
-  PTextToSpeech * tts = new PTextToSpeech();
-  tts->SetEngine(tts->GetEngines()[0]);
+  PTextToSpeech * tts = NULL;
+  PFactory<PTextToSpeech>::KeyList_T engines = PFactory<PTextToSpeech>::GetKeyList();
+  if (engines.size() != 0) {
+    PString name;
+#ifdef _WIN32
+    name = "Microsoft SAPI";
+    if (std::find(engines.begin(), engines.end(), name) == engines.end())
+#endif
+      name = engines[0];
+    tts = PFactory<PTextToSpeech>::CreateInstance(name);
+  }
   SetTextToSpeech(tts, TRUE);
 }
diff -Nur openivr-1.13.5/version.h openivr/version.h
--- openivr-1.13.5/version.h	2003-06-06 08:10:59.000000000 +0200
+++ openivr/version.h	2005-03-11 01:43:39.000000000 +0100
@@ -29,8 +29,6 @@
  *
  * Library dependencies:
  *
- *   pwlib: v1.5.0   CVS tag:
- *   openh323: v1.12.0   CVS tag: v1_12_0
  */
 
 #ifndef _OpenIVR_VERSION_H
@@ -39,7 +37,7 @@
 #define MAJOR_VERSION 1
 #define MINOR_VERSION 0
 #define BUILD_TYPE    ReleaseCode
-#define BUILD_NUMBER 6
+#define BUILD_NUMBER 7
 
 
 #endif  // _OpenIVR_VERSION_H
